src/classes/CON_Email.cls

```diff
@@ -0,0 +1,132 @@
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+/**
+* @author Salesforce.org
+* @date 2016
+* @group Contacts
+* @group-content ../../ApexDocContent/Contacts.htm
+* @description A Class to handle Contact Email related Items
+*
+*/
+public class CON_Email {
+
+    /*******************************************************************************************************
+    * @description Sets the Email field depending on the value of Preferred Email
+    * @param Contact the contact to update
+    */
+	public static void processPreferredEmail(Contact contact, List<Contact> oldlist) {
+
+		Boolean isUpdate =  (oldlist != null) ? true : false;
+
+	    // Store default email field mappings to preferred email pick list.
+	    Map<String,String> preferredEmailMappings = new Map<String,String>{   
+	        'University' => 'UniversityEmail__c',
+	        'Work' => 'WorkEmail__c',
+	        'Alternate' => 'AlternateEmail__c'
+	    };
+
+	    Map<String,String> valuedEmailMap = emailFieldsWithValue(contact);
+
+        if( valuedEmailMap.size() > 0 ) {
+
+            // Enforce preferred email field
+            if(String.isBlank(contact.Preferred_Email__c)){
+                contact.addError( Label.PreferredEmailRequiredError );
+            }
+
+            // Get the Api name that matches the value in Preferred Email
+            String preferredApiField = ( preferredEmailMappings.containsKey(contact.Preferred_Email__c) ) ? preferredEmailMappings.get(contact.Preferred_Email__c) : valuedEmailMap.get(contact.Preferred_Email__c);
+
+            String prefEmailVal = (String)contact.get(preferredApiField);
+
+            if (  String.isNotBlank( prefEmailVal ) ) { 
+                contact.Email = prefEmailVal;
+            } else {
+                contact.addError(Label.PreferredEmailMatchNotNull);
+            }      
+        
+        } else {
+
+            // Cannot have preferred email set if there are no emails present
+            if( !String.isBlank(contact.Preferred_Email__c)) {
+                contact.addError(Label.PreferredEmailMatchNotNull);                    
+            }
+
+            // oldlist is NULL onInsert so we check to make sure this logic only applies on updates.
+			if( isUpdate && String.isNotBlank(contact.Email) ) {
```

Comment:
We are not using this class in the delete context, only insert/update. However, I agree checking to make sure contact isn't null will make the method a little more reusable/future proof